lib/util/ios-proxy.js

```diff
@@ -91,73 +101,82 @@ let iOSProxy = {
     }
   }),
 
-  checkIfRunning() {
+  checkIfRunning: task.async(function*() {
     // Check for any currently running instances of ios_webkit_debug_proxy.
     let running = false;
-    let cmd = "/usr/bin/pgrep";
-    let name = "ios_webkit_debug_proxy";
-    let args = ["-x"];
+    let cmd = "/usr/bin/pgrep -x ";
+    let name = binaryName;
     switch(Runtime.OS) {
       case "Linux": // fall-through
         // "pgrep -x" only matches against the first 15 characters on Linux.
         name = name.slice(0, 15);
       case "Darwin":
-        args.push(name);
+        cmd += name;
+        break;
+      case "WINNT":
+        // "ps" only matches the file name without extension on Windows.
+        name = name.slice(0, -4);
+        cmd = `PowerShell -Command ps ${name}`;
         break;
       default:
-        // XXX: how do we do this on Windows?
-        return Promise.resolve(false);
+        throw new Error("iOS proxy not yet supported on " + Runtime.OS);
     }
 
-    return new Promise((resolve, reject) => {
-      let proc = child_process.spawn(cmd, args);
-      proc.stdout.on("data", data => {
-        console.log(cmd + ": " + data);
-        if (!Number.isNaN(parseInt(data))) {
-          running = true;
-          // XXX: notification that an instance is already running?
-        }
-      });
-
-      proc.stderr.on("data", data => console.error(cmd + ": " + data));
-
-      proc.on("close", code => {
-        console.log(cmd + " exited with code " + code);
-        resolve(running);
-      });
-    });
-  },
+    return this.runInShell(cmd);
+  }),
 
   spawn() {
-    proxyProcess = child_process.spawn(this.binaryPath,
-                                   ["-c", "null:9221,:9230-9240", "-F"],
-                                   {
-                                     env: this.environment
-                                   });
+    let devicePortConfig = "null:9221,:9230-9240";
+    let args = ["-c", devicePortConfig, "--no-frontend"];
+    proxyProcess = child_process.spawn(this.binaryPath, args, this.options);
     proxyProcess.stdout.on("data", makeInfallible(function (data) {
-      console.log("ios_webkit_debug_proxy: " + data);
+      console.log(binaryName + ": " + data);
     }, "proxyProcess.stdout.on callback"));
 
     proxyProcess.stderr.on("data", makeInfallible(function (data) {
-      console.error("ios_webkit_debug_proxy: " + data);
+      console.error(binaryName + ": " + data);
     }, "proxyProcess.stderr.on callback"));
 
     proxyProcess.on("close", makeInfallible(function (code) {
-      console.log("ios_webkit_debug_proxy exited with code " + code);
+      console.log(binaryName + " exited with code " + code);
     }, "proxyProcess.on callback"));
 
     unload(() => proxyProcess && proxyProcess.kill());
   },
 
   checkIfBound: task.async(function*() {
+    if (Runtime.OS == "WINNT") {
+      // Windows has no reasonably simple way to make this work.
+      return Promise.resolve(true);
```

Comment:
Something like `netstat -an | find "0.0.0.0:<PORT>" | find "LISTENING"` could be workable I think.